packages/twenty-server/src/modules/workflow/workflow-trigger/services/workflow-trigger.workspace-service.ts

```diff
@@ -19,26 +31,43 @@ export class WorkflowTriggerWorkspaceService {
   constructor(
     private readonly twentyORMManager: TwentyORMManager,
     private readonly workflowCommonWorkspaceService: WorkflowCommonWorkspaceService,
-    private readonly workflowRunnerWorkspaceService: WorkflowRunnerWorkspaceService,
+    private readonly workflowStatusWorkspaceService: WorkflowStatusWorkspaceService,
+    @InjectMessageQueue(MessageQueue.workflowQueue)
+    private readonly messageQueueService: MessageQueueService,
+    private readonly scopedWorkspaceContextFactory: ScopedWorkspaceContextFactory,
   ) {}
 
   async runWorkflowVersion(workflowVersionId: string, payload: object) {
+    const workspaceId = this.scopedWorkspaceContextFactory.create().workspaceId;
+
+    if (!workspaceId) {
+      throw new WorkflowTriggerException(
+        'No workspace id found',
+        WorkflowTriggerExceptionCode.INTERNAL_ERROR,
+      );
+    }
+
     const workflowVersion =
       await this.workflowCommonWorkspaceService.getWorkflowVersion(
         workflowVersionId,
       );
 
-    try {
-      return await this.workflowRunnerWorkspaceService.run({
-        action: workflowVersion.trigger.nextAction,
-        payload,
-      });
-    } catch (error) {
+    if (!workflowVersion) {
       throw new WorkflowTriggerException(
-        `Error running workflow version ${error}`,
-        WorkflowTriggerExceptionCode.INTERNAL_ERROR,
+        'No workflow version found',
+        WorkflowTriggerExceptionCode.INVALID_WORKFLOW_VERSION,
       );
     }
+
+    return await this.startWorkflowRun(
+      workspaceId,
+      workflowVersionId,
+      payload,
+      {
+        source: FieldActorSource.MANUAL,
+        name: '', // TODO: Add name
```

Comment:
can't you get the firstName lastName from the auth token to compute a name here? Like it is done in `packages/twenty-server/src/engine/metadata-modules/field-metadata/query-hooks/created-by.pre-query-hook.ts`.L80